### PR TITLE
Issue 38

### PR DESCRIPTION
**Pull request by FrancisToth**
This PR:
- Fixes a stackoverflow error in HelpDoc
- Implements the HelpDoc.toHTML method
- Provides some basic CSS to render the HelpDoc in HTML

### CODE REVIEW

**Review by jdegoes — APPROVED** [review]
Would be nice to see a screenshot, but looks good to me. Thank you!

File: zio-cli/shared/src/main/scala/zio/cli/HelpDoc.scala [hunk 1]
```diff
@@ -68,6 +68,142 @@ sealed trait HelpDoc { self =>
       case x                             => x
     }
 
+  def toHTML(): String = {
```
Comment (jdegoes) [inline zio-cli/shared/src/main/scala/zio/cli/HelpDoc.scala:71]: Ah, because this is a pure function, can you drop the `()` after `toHtml`?

Comment (FrancisToth) [reply zio-cli/shared/src/main/scala/zio/cli/HelpDoc.scala]: Done :)

**Comment by FrancisToth** [comment]
@jdegoes :  Here's a [screenshot](https://raw.githubusercontent.com/FrancisToth/zio-cli/screenshot/helpdoc.png) generated by outputting the HTML of the following `HelpDoc`

```scala
      HelpDoc.h1(HelpDoc.Span.text("Title (h1)")) +
      HelpDoc.p(HelpDoc.Span.text("Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.")) +
      HelpDoc.h2(HelpDoc.Span.text("A Title (h2)")) +
      HelpDoc.p(HelpDoc.Span.text("Contrary to popular belief, Lorem Ipsum is not simply random text. It has roots in a piece of classical Latin literature from 45 BC, making it over 2000 years old. Richard McClintock, a Latin professor at Hampden-Sydney College in Virginia, looked up one of the more obscure Latin words, consectetur, from a Lorem Ipsum passage, and going through the cites of the word in classical literature, discovered the undoubtable source. Lorem Ipsum comes from sections 1.10.32 and 1.10.33 of \"de Finibus Bonorum et Malorum\" (The Extremes of Good and Evil) by Cicero, written in 45 BC. This book is a treatise on the theory of ethics, very popular during the Renaissance. The first line of Lorem Ipsum, \"Lorem ipsum dolor sit amet..\", comes from a line in section 1.10.32.")) +
      HelpDoc.h3(HelpDoc.Span.text("A Title (h3)")) +
      HelpDoc.p(HelpDoc.Span.text("It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout. The point of using Lorem Ipsum is that it has a more-or-less normal distribution of letters, as opposed to using 'Content here, content here', making it look like readable English. Many desktop publishing packages and web page editors now use Lorem Ipsum as their default model text, and a search for 'lorem ipsum' will uncover many web sites still in their infancy. Various versions have evolved over the years, sometimes by accident, sometimes on purpose (injected humour and the like).")) +
      HelpDoc.p(HelpDoc.Span.code("""
      |package zio
      |
      |import zio.internal.Platform
      |
      |trait BootstrapRuntime extends Runtime[ZEnv] {
      |  val environment: ZEnv = ZEnv.Services.live
      |
      |  /**
      |   * The platform of the runtime, which provides the essential capabilities
      |   * necessary to bootstrap execution of tasks.
      |   */
      |  val platform: Platform = Platform.default
      |}
      |""".stripMargin))
```

**Comment by jdegoes** [comment]
Looks nice!

**Pull request merged**